docs/content/05_tutorials/github-prs.md

```diff
@@ -0,0 +1,493 @@
+
+# Specifying the model
+
+In a Prodo app your state is stored in a global store which is accessed by your
+components and updated by your actions. The type of this state is defined in a
+[model](/basics/model) in `src/model.ts`. Plugins can also be added to the model to extend the
+functionality.
+
+The model is created with `createModel` using the `State` type. Initially the
+app has not state. We also add the [logger](/plugins/logger) and
+[route](/plugins/routing) plugins. The logger plugin logs useful information to
+the console. The route plugin gives us client side routing.
+
+## Adding some state
+
+In our app we want to store a list of pull requests per repo. We can do that with:
+
+```ts
+import { createModel } from "@prodo/core";
+import loggerPlugin from "@prodo/logger";
+import routePlugin from "@prodo/route";
+
+// highlight-start
+export interface PullRequest {
+  id: number;
+  prNumber: number;
+  title: string;
+  url: string;
+  author: string;
+  authorImage: string;
+  createdAt: string;
+}
+
+export interface State {
+  pullRequests: { [key: string]: PullRequest[] };
+}
+// highlight-end
+
+export const model = createModel<State>()
+  .with(loggerPlugin)
+  .with(routePlugin);
+
+export const { state, watch, dispatch } = model.ctx;
+```
+
+The list of pull requests will be keyed by a string that matches the
+`owner/repo` format.
+
+When creating the store in `src/index.tsx` we now need to initialize the
+`pullRequests` in our state.
+
+```ts
+// ...
+
+const history = createBrowserHistory();
+const { Provider } = model.createStore({
+  logger: true,
+  route: {
+    history,
+  },
+  // highlight-start
+  initState: {
+    pullRequests: {},
+  },
+  // highlight-end
+});
+
+// ...
+```
+
+# Routing
+
+Our app will have two routes, one for searching and one for a repo. CPA includes
+a pre-configured [route plugin](/plugins/routing) so all we need to do is define
+the routes in `src/App.tsx`.
+
+```tsx
+import * as React from "react";
+import { Switch, Route } from "@prodo/route";
+import Header from "./components/Header";
+import NotFound from "./pages/NotFound";
+// highlight-start
+import Home from "./pages/Home";
+import Repo from "./pages/Repo";
+// highlight-end
+
+const App = () => (
+  <div className="app">
+    <Header />
+
+    <div className="container">
+      <Switch>
+        // highlight-start
+        <Route exact path="/" component={Home} />
+        <Route path="/:owner/:repo" component={Repo} />
+        // highlight-end
+        <Route component={NotFound} />
+      </Switch>
+    </div>
+  </div>
+);
+
+export default App;
+```
+
+Route parameters get passed to components as props. Create a _Repo_ page in
+`src/pages/Repo.tsx` with the following contents:
+
+```tsx
+import * as React from "react";
+import { Link } from "@prodo/route";
+
+export interface Props {
+  owner: string;
+  repo: string;
+}
+
+const Repo: React.FC<Props> = ({ owner, repo }) => (
```

Comment:
Is it worth explaining React.FC?

Reply:
I don't think so, we are assuming they know React